### PR TITLE
Upgrade restrictions

### PR DESCRIPTION
**Pull request by Kucharssim**
fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1647

This PR is a copy of https://github.com/jasp-stats/jaspAnova/pull/146 and adds two additional fixes to bugs found by Rebecca. The original PR was merged into a separate branch in order to produce a special nightly for a GORICA workshop. But we need to merge it to master so that it appears in the next release.

### CODE REVIEW

**Comment by Kucharssim** [comment]
I could have squashed the commits, oh well... 🤔

File: tests/testthat/test-anovarepeatedmeasures.R [hunk 1]
```diff
@@ -344,141 +344,6 @@ test_that("Analysis handles errors", {
 
 })
 
-# Order restrictions
-
-test_that("Model Comparison Table results match", {
-  options <- initOpts()
-  options$restrictedModels <- list(list(restrictionSyntax = "DrinkBeer.ImageryPositive == DrinkBeer.ImageryNegative", modelName = "Model 1",
```
Comment (vandenman) [inline tests/testthat/test-anovarepeatedmeasures.R:351]: why did you delete these tests?

**Review by vandenman — CHANGES_REQUESTED** [review]
The changes look sensible to me, aside from deleting a set of unit tests.

File: tests/testthat/test-anovarepeatedmeasures.R (hunk 1 above)
Comment (Kucharssim) [reply tests/testthat/test-anovarepeatedmeasures.R:351]: The rm anova restrictions and an(c)ova restrictions now use the same functions. For an(c)ova, I also made new tests using the examples on the https://restriktor.org/ website, so the majority of the restrictions code is tested there for consistency with the restriktor package. In the tests for rm anova, I focused on making tests that trigger code that is unique to rm anova.

The rewrite made the input and output structure different, so it was easier and less error prone to make new tests rather than modifying the old ones. I could reinstate these tests and fix them to work with the rewrite, but it seemed like an effort that does not add much.

**Review by vandenman — APPROVED** [review]

**Comment by vandenman** [comment]
@Kucharssim feel free to merge once you're sure you haven't forgotten any upgrades :wink:
